### ISSUE TITLE
PPO recipe improvements v.2.1

### ISSUE THREAD

**Issue by SalmanMohammadi** [pytorch/torchtune#2061]
- We can re-use the same `setup_models` for all of the models, rather than setup each model individually inside `setup_models`
- I personally think we should disable gradient accumulation. DeepspeedChat doesn't support it whilst TRL does. However, we already offer a `ppo_batch_size` arg to control minibatch size and the added gradient accumulation logic makes the code harder to interpret.
- Only use the last valid non-padding token to estimate the scores from the reward model during trajectory estimation, rather than the entire truncated sequence.
